version: '3'

services:
  mbp_chatbot_db:
    image: postgres:latest
    restart: always
    environment:
      - POSTGRES_USER=postgres
      - POSTGRES_PASSWORD=postgres
    ports:
      - '5434:5432' # Changer le port exposé pour éviter le conflit
    volumes:
      - mbp_chatbot_db:/var/lib/postgresql/data
    networks:
      - rabbitmq_go_net

  rabbitmq:
    image: rabbitmq:3-management-alpine
    container_name: 'mbp-chatbot-rabbitmq'
    ports:
      - 5674:5672 # Changer le port exposé pour éviter le conflit
      - 15674:15672 # Interface de gestion avec un port différent
    volumes:
      - ~/.docker-conf/rabbitmq/app2-data/:/var/lib/rabbitmq/
      - ~/.docker-conf/rabbitmq/app2-log/:/var/log/rabbitmq
    networks:
      - rabbitmq_go_net
  bot-app:
    build: .
    ports:
      - '3001:3001'
    depends_on:
      - mbp_chatbot_db
      - mbp-chatbot-rabbitmq
    environment:
      - DATABASE_URL=postgresql://postgres:postgres@mbp_chatbot_db:5434/mbp_chatbot_db
      - RABBITMQ_URL=amqp://mbp-chatbot-rabbitmq:5674
    networks:
      - rabbitmq_go_net

volumes:
  mbp_chatbot_db:
    driver: local

networks:
  rabbitmq_go_net:
    driver: bridge
